tests/demo/scene_sun_temple.hpp

```diff
@@ -0,0 +1,221 @@
+
+#pragma once
```

Comment:
This is not a maintainable way to add actual scenes. This system is more meant for test scenes. if we want to do save scenes like this we should write a proper system for it. This makes it far to unmaintainable. 1 change and you need to change a lot of scenes. I'll probably remove this when we have such a system.